### ISSUE TITLE
[Feature Request] Ability to disable the "delete remote branch" checkbox in repo settings

### ISSUE THREAD

**Issue by geordiemhall** [ForkIssues/TrackerWin#652]
I personally don't really need this feature, but a few artists in our office have recently switched over to Fork from SourceTree, and have told me they get anxious about accidentally deleting a remote branch along with their local branch since the checkbox is quite prominent in the delete dialog. 

![image](https://user-images.githubusercontent.com/3511383/74115359-00c8e480-4c03-11ea-83cc-535e164e92e0.png)

The ideal solution for this is probably a setting on the remote server to forbid deleting of remote branches via a git client, to eg. force you to delete it through the web client or via pull request being merged (after which you could then delete your local branch).

But as a client-side safeguard, it could be nice to have a per-repo option in Fork to always disable that checkbox so that they never have to worry about clicking it accidentally :)

**Comment by felixhirt** [comment]
or maybe make two separate buttons (delete local, delete local and remote), where one has a big red exclamation mark?

**Comment by JeReT** [comment]
I usually delete branches which have been merged and deleted after a merge request. In the new version of fork these branches are easily identifiable (they have a yellow icon indicating that the tracking reference got lost).
I think the warning should be suppressed if the branch lost its tracked reference and there are no local commits in the branch. 

Can you confirm whether you are also usually deleting branches with these conditions?

**Comment by geordiemhall** [comment]
Yeah usually I'd only delete the local after the remote has been deleted first too.

Though I can also see there being cases where you're handing off work to someone else, so you don't want to delete the remote branch yet, just clean up your local branches. In which case it's still quite easy to accidentally delete both and hose all of that work.

So in the spirit of just making it hard to do something very destructive unintentionally, even just an extra confirmation dialog saying "You're going to delete the remote branch as well, are you sure?" would probably suffice. 

I think SourceTree might even make the warning more prominent when the remote branch hasn't been merged into your default branch yet too.

**Comment by felixhirt** [comment]
I agree that there should be considerable barriers (ideally a warning message) when deleting a remote branch with unmerged changes. I think Fork already does something like this for unmerged local branches

**Comment by jerone** [comment]
> The ideal solution for this is probably a setting on the remote server to forbid deleting of remote branches via a git client

Azure DevOps provides this information with an [`isLocked` property](https://docs.microsoft.com/en-us/rest/api/azure/devops/git/refs/list?view=azure-devops-rest-4.1#gitref).

**Comment by geordiemhall** [comment]
@jerone ah nice, wonder if BitBucket has something similar. Last I checked you could specify "protected" branches, but didn't seem like there was something that would apply to all new branches by default
